automation-tests/test/scratchOrg.ts

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import child_process from 'child_process';
+import fs from 'fs';
+import path from 'path';
+import util from 'util';
+import {
+  DefaultTreeItem,
+  InputBox,
+  QuickOpenBox,
+  Workbench
+} from 'wdio-vscode-service';
+import {
+  EnvironmentSettings
+} from './environmentSettings';
+import {
+  utilities
+} from './utilities';
+
+const exec = util.promisify(child_process.exec);
+
+export class ScratchOrg {
+  private testSuiteSuffixName: string = undefined;
+  private reuseScratchOrg = false;
+  private projectFolderPath: string = undefined;
+  private prompt: QuickOpenBox | InputBox = undefined;
+  private scratchOrgAliasName: string = undefined;
+
+  public constructor(testSuiteSuffixName: string, reuseScratchOrg: boolean) {
+    this.testSuiteSuffixName = testSuiteSuffixName;
+    this.reuseScratchOrg = reuseScratchOrg;
+
+    // To have all scratch orgs be reused, uncomment the following line:
```

Comment:
Is manually overriding this desirable given the constructor also takes it as a boolean?

Reply:
This overrides globally, for all suites.  When the value passed to the ctor() is changed, it's only for that suite.

Reply:
Should that maybe be another setting in `EnvironmentSettings` instead of having it be in the class?